dev/tasks/python-minimal-build/github.linux.yml

```diff
@@ -26,7 +26,7 @@ jobs:
     runs-on: ubuntu-latest
 {{ macros.github_set_env(env) }}
     steps:
-      {{ macros.github_checkout_arrow(submodules=false)|indent }}
+      {{ macros.github_checkout_arrow(fetch_depth=0, submodules=false)|indent }}
```

Comment:
@raulcd I'm just curious! Why was this change needed?

Reply:
The default fetch depth  for `actions/checkout` is 1. That means it will checkout the specific commit and not fetch anything else, this is useful when dealing with large repos to speed up the checkout but it also leaves out all other branches and more importantly in this case tags (which are needed to calculate the dev version). 0 will fetch the full repo so all tags and branches are available.

Reply:
Yes, that's exactly the case. I realized while testing that the version was not generated correctly due to the missing tags, adding `fetch_depth=0` pulls the tags and generates the correct version for pyarrow.